packages/playground/blueprints/src/lib/steps/activate-plugin.ts

```diff
@@ -30,10 +33,30 @@ export const activatePlugin: StepHandler<ActivatePluginStep> = async (
 			`Required WordPress files do not exist: ${requiredFiles.join(', ')}`
 		);
 	}
+
 	await playground.run({
 		code: `<?php
-      ${requiredFiles.map((file) => `require_once( '${file}' );`).join('\n')}
-      activate_plugin('${pluginPath}');
-      `,
+${requiredFiles.map((file) => `require_once( '${file}' );`).join('\n')}
+$plugin_path = '${pluginPath}';
+$plugin_entry_file = $plugin_path;
+if (is_dir($plugin_path)) {
+	// Find plugin entry file
+	$plugin_entry_file = '';
+	$files = glob( $plugin_path . '/*.php' );
```

Comment:
this already returns an empty list if there are no matches. `false` if there was an error.

this isn't that big of a deal, but we could probably rewrite this and void the nesting of the `foreach`

```php
$files = glob( … );
if ( false === $files ) {
	// Do we even need to handle this?
}

foreach( $files as $file ) {

}
```

if we don't think that will error out, because we expect the `$plugin_path` to always exist, then we can skip over the middleman. it's already taken care of above by `is_dir()`

```php
foreach ( glob( $plugin_path . '/*.php' ) as $file ) {
	// stuff
}
```

but we can also avoid the race condition (as unlikely as we are to trigger it) since `glob()` already handles the `is_dir()` check internally

```php
$plugin_path = '${pluginPath}';

foreach ( ( glob( $plugin_path . '/*.php' ) ?: array() ) as $file ) {
	$info = …
	if ( ! empty( $info['Name'] ) ) {
		activate_plugin( $file );
		return;
	}
}

throw new Exception('Could not find plugin entry file.');
```

Reply:
Ah thank you, that is simpler code. Returning early, without use of a second variable.

For backward compatibility with existing code using the `activatePlugin` function, it accepts either:

- Plugin entry file relative to the plugins directory: `plugin-name/plugin-name.php`
- Path to the plugin directory: `/wordpress/wp-content/plugins/plugin-name`

The first is what the WP function [activate_plugin](https://developer.wordpress.org/reference/functions/activate_plugin/) expects, and it requires the caller to find the entry file themselves.

The second is the new behavior which finds it for you. To support both cases:

```php
if (!is_dir($plugin_path)) {
	activate_plugin($plugin_path);
	return;
}
// Find plugin entry file
foreach ( ( glob( $plugin_path . '/*.php' ) ?: array() ) as $file ) {
	$info = get_plugin_data( $file, false, false );
	if ( ! empty( $info['Name'] ) ) {
		activate_plugin( $file );
		return;
	}
}
throw new Exception('Could not find plugin entry file.');
```